### PR TITLE
Paint setup crooked house

### PR DESCRIPTION
**Pull request by marijnvdwerf**
image

### CODE REVIEW

File: src/ride/track_paint.c [hunk 1]
```diff
@@ -990,3 +991,326 @@ TRACK_PAINT_FUNCTION get_track_paint_function_facility(int trackType, int direct
...
+			rct_g1_element *sprite = &g1Elements[ride->vehicles[1]];
+			RCT2_GLOBAL(RCT2_ADDRESS_PAINT_SETUP_CURRENT_TYPE, uint8) = VIEWPORT_INTERACTION_ITEM_SPRITE;
+			RCT2_GLOBAL(0x9DE578, rct_g1_element *) = sprite;
+		}
+	}
+
+
+	uint32 image_id = (direction + ride_type->vehicles[0].base_image_id) | RCT2_GLOBAL(0x00F441A0, uint32);
+
+	rct_crooked_house_bound_box esi = crooked_house_data[segment];
```
Comment (duncanspumpkin) [inline src/ride/track_paint.c]: can you use a better var name than `esi`

File: src/ride/track_paint.c [hunk 2]
```diff
@@ -990,3 +991,326 @@ TRACK_PAINT_FUNCTION get_track_paint_function_facility(int trackType, int direct
...
+	RCT2_GLOBAL(0x14281B0, uint32) = segment;
+	RCT2_GLOBAL(0x14281B4, uint32) = direction;
+
+	rct_ride *ride = get_ride(original_map_element->properties.track.ride_index);
+
+	rct_ride_entry *ride_type = get_ride_entry(ride->subtype);
+
+	RCT2_GLOBAL(0x0014281B8, uint32) = ride_type->vehicles[0].base_image_id;
+
+	uint32 ebx = 0xFFFFFFFF;
```
Comment (duncanspumpkin) [inline src/ride/track_paint.c]: Variable not used.

File: src/ride/track_paint.c (hunk 1 above)
Comment (marijnvdwerf) [inline src/ride/track_paint.c]: Ah yes. That was from before we knew what those variables actually were.

File: src/ride/track_paint.c [hunk 3]
```diff
@@ -990,3 +991,326 @@ TRACK_PAINT_FUNCTION get_track_paint_function_facility(int trackType, int direct
...
+	if (ride->lifecycle_flags & RIDE_LIFECYCLE_ON_TRACK) {
+		if (ride->vehicles[0] != 0xFFFF && ride->vehicles[1] != 0xFFFF) {
```
Comment (duncanspumpkin) [inline src/ride/track_paint.c]: Only checks the first vehicle not the second. `movsx ebx, WORD ride->vehicles`

File: src/ride/track_paint.c [hunk 4]
```diff
@@ -990,3 +991,326 @@ TRACK_PAINT_FUNCTION get_track_paint_function_facility(int trackType, int direct
...
+			rct_g1_element *sprite = &g1Elements[ride->vehicles[1]];
```
Comment (duncanspumpkin) [inline src/ride/track_paint.c]: vehicle 0

**Comment by janisozaur** [comment]
```
/home/janisozaur/workspace/OpenRCT2/src/ride/gentle/crooked_house.c:165:25: warning: comparison of constant -1 with expression of type 'uint16'
      (aka 'unsigned short') is always true [-Wtautological-constant-out-of-range-compare]
                if (ride->vehicles[0] != -1) {
                    ~~~~~~~~~~~~~~~~~ ^  ~~
1 warning generated.
```

**Pull request merged**